simp/rubygem-simp-cli
# (SIMP-1448) Rework simp config for SIMP 6

- Output hieradata appropriate for SIMP6 (correct
  names and types).
- Use scenario yaml file as a beginning point for parameter
  values and only prompt for what is missing.
- Set $simp_scenario variable in 'simp' environment's site.pp
- Only output global SIMP hieradata in simp_config_settings.yaml.
- Set SIMP server hieradata in the server's <host>.yaml file,
  instead of SIMP's global hieradata file.
- Copy simp from /usr/share/simp if copy has not yet been done
  for NON-ISO, RPM install.  NOTE:  Does NOT prompt user,
  as simp config operation won't work if those files are not
  in place. SIMP-1561
- Ask user if they want to set simp to the default production
  environment and then create production links to simp in the
  primary and (SIMP-specific) secondary Puppet environments
  directories.
- Rework logic relating to LDAP to distinguish between the
  case in which LDAP server is or is not on the SIMP server.
  - SIMP-647
  - Remove scraping of openlap configuration files to determine
    OS values for LDAP Base DN, Bind DN, RootDN, and server URIs.
    This method of determining LDAP config is unreliable.
- Rework logic related to YUM repos, so that potential problems
  with external repos are highlighted. SIMP-647
  - Detect when external repos are required and setup as
    simp::yum parameters as much as we can
  - Force user to verify, manually, that external repos are
    setup before 'simp bootstrap' can run
- Use current FIPS mode to determine simp_options::fips setting
  instead of giving user the option from 'simp config' to screw
  up their system.
- Bug fixes
  - Don't output text from grub2 configuration, certificate
    creation, and yum repo update SIMP-407, SIMP-409, SIMP-410
  - For ISO/RPM installs (only), handle the problem
    in SIMP-1416. Solution for R10k systems TBD.
  - Don't log intro and summary text when --quiet option is
    specified.
- Minor enhancements for end-user
  - Add log of simp config output, including details of the apply
  - Capture and log error output from spawned processes
    (primarily from ActionItems)
  - Suppress ActionItem non-error console output
  - Backup more of the files modified by simp config and include
    a timestamp in those file names .
  - Use start time of 'simp config' for any backups, so the user
    can correlate all changes 'simp config' made in a run.
  - Clean up text presented to end user
    - Improve output from ActionItems
    - Improve Item descriptions/summaries
    - Make sure PasswordItems tell you which password they are
      querying for, when --quiet mode is selected
  - Write keys in alphabetical order into answers YAML file and
    global hieradata YAML file, so that the user can easily navigate
    these documents.
  - Don't ask about forwarding syslog servers if no syslog servers
    are specified.
- Minor refactoring for maintainability
  - Add IntegerItem to support integer hieradata.
  - Only implement apply in action items.  Allows us to
    use ActionItem.description to describe tersely what is
    happening while it is happening.  (Warm fuzzy for user.)
  - Rename Config::Items for clarity/maintainability
  - Clean up options list for simp config
  - Implement error reporting for Item dependencies.  This makes
    debugging Item tree errors easier.
  - Rework directory layout for Items to facilitate code maintenance.
- Additional tests

SIMP-407 #close
SIMP-409 #close
SIMP-410 #close
SIMP-496 #close
SIMP-647 #close
SIMP-1191 #close
SIMP-1448 #close
SIMP-1540 #close
SIMP-1561 #close
